### ISSUE TITLE
Training model from scratch results in incompatibility with the testing script and inferior visual quality after adjustments

### ISSUE THREAD

**Issue by albek00** [xiezhy6/PASTA-GAN#12]
I have attempted to reproduce high levels of quality shown in your pretrained model through training on a substantively expanded dataset. However, I have encountered a number of divergences between your dedicated training and testing dataset classes (UvitonDatasetFull and UvitonDatasetV19_test). Notably, training dataset, network architecture and, consequently, all new models trained through the provided script use two arrays of normalized body part images (norm_img and norm_img_lower, with their corresponding shapes at 30x64x64 and 12x64x64) as style encoding inputs, while testing dataset and pretrained model use only one array of normalized body part images, concatenated with a normalized pose representation (norm_img and norm_pose or ‘stickman’, with shapes at 30x64x64 and 30x64x64).

My approach to resolving these differences so far was based on bringing the training dataset class into closer alignment with the testing dataset through modification of its normalize() function. After these dataset adjustments and slight changes to the training script to accommodate the new input flow, I was able to train a network model with style encoding input shape (norm_img and norm_pose) at 60x64x64, fully in line with the pretrained model. However, as attached images show, the resulting level of visual quality is far removed from yours, even at relatively advanced stages of training (8000-12000 iterations). Notably, shoulder area and general body shape experience unexpected deformations relative to the original pose, while use of full-body images as either person or garment causes severe distortion.

Given this disappointing outcome, would it be possible for you to provide some feedback to the general direction of my efforts? Is there something I might have overlooked in my attempts to bring two dataset classes in line? Were there any additional parameters in the training script that should have received more of my attention?

![1a95a642b4a8440a81c96cc928b40e1c__5c5b36b78f2341e48946768b5c490ef9](https://user-images.githubusercontent.com/102261725/159814768-45d552f9-db5b-468a-8b41-2a235deb6929.png)
![1a30787ed8a94293a2ba2a6b0eb0aa32__0f47cb9cb50a4dba8bcdaa68ae4da54d](https://user-images.githubusercontent.com/102261725/159814885-26ef8e3b-439e-4b8b-b9c7-e9fa8e86b4ce.png)
![1a30787ed8a94293a2ba2a6b0eb0aa32__2ec33e777b9d4905941b1b33d670045a](https://user-images.githubusercontent.com/102261725/159814917-a7f43df9-942f-4489-8299-d6ea71d17906.png)
![2d1c8a0485704f1291c6a02be96bdac8__1a7a5302159e48af84a690e3ad456489](https://user-images.githubusercontent.com/102261725/159815105-08e86945-de12-4293-8a9c-6b93e496a325.png)
![2d1c8a0485704f1291c6a02be96bdac8__0bc002dc7e474daeb9a97688043fd83f](https://user-images.githubusercontent.com/102261725/159815132-359cffd6-c2e4-467d-9c67-9f2b6ad7b46f.png)
![6b9c0d92645049bc9906c7d418c1f09c__4d21f15280db465db0169b5804620d64](https://user-images.githubusercontent.com/102261725/159815355-20274f22-bbc8-4309-8780-36a77163d5e8.png)
![6b9c0d92645049bc9906c7d418c1f09c__1ec58bc1d00a4542bddd59c03664b24e](https://user-images.githubusercontent.com/102261725/159815372-c9efc059-10a8-49a1-a8ee-51c7b449420e.png)
